### PR TITLE
Add APIs to read GlobalIP for GN2.0 objects

### PR DESCRIPTION
**Pull request by sridhargaddam**
This PR implements APIs that return the GlobalIPs part of ClusterGlobalEgressIP
as well as GlobalIngressIP objects.

Signed-Off-by: Sridhar Gaddam <sgaddam@redhat.com>
Co-authored-by: Vishal Thapar <5137689+vthapar@users.noreply.github.com>

### CODE REVIEW

File: test/e2e/framework/clusterglobalegressip.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+				if result == nil {
+					return false, fmt.Sprintf("ClusterGlobalEgressIP %s not found yet", name), nil
+				}
+
+				cgeipUnstruct := result.(*unstructured.Unstructured)
+				globalIPs, found, err := unstructured.NestedStringSlice(cgeipUnstruct.Object, "status", "allocatedIPs")
+				if err != nil || !found || len(globalIPs) == 0 {
+					return false, "", fmt.Errorf("ClusterGlobalEgressIP %q exists but allocatedIPs not available yet",
+						cgeipUnstruct)
+				}
```
Comment (tpantelis) [inline test/e2e/framework/clusterglobalegressip.go]:
```suggestion
				globalIPs := getGlobalIPs(result.(*unstructured.Unstructured))
				if len(globalIPs) == 0 {
					return false, fmt.Sprintf("ClusterGlobalEgressIP %q exists but allocatedIPs not available yet", name), nil
				}
```

File: test/e2e/framework/globalingressips.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+				if result == nil {
+					return false, "globalip not found yet", nil
+				}
+
+				gipUnstruct := result.(*unstructured.Unstructured)
+				globalIP, found, err := unstructured.NestedString(gipUnstruct.Object, "status", "allocatedIP")
+				if err != nil || !found || globalIP == "" {
+					return false, "", fmt.Errorf("GlobalIngress %q exists but allocatedIP not available yet",
+						gipUnstruct)
+				}
```
Comment (tpantelis) [inline test/e2e/framework/globalingressips.go]:
```suggestion
				globalIP := getGlobalIP(result.(*unstructured.Unstructured))	
				if globalIP == "" {
					return false, fmt.Sprintf("GlobalIngress %q exists but allocatedIP not available yet",
						name), nil
				}
```

File: test/e2e/framework/globalingressips.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+			func() (interface{}, error) {
+				resGip, err := gipClient.Get(context.TODO(), name, metav1.GetOptions{})
+				if apierrors.IsNotFound(err) {
+					return nil, nil
+				}
+				return resGip, err
+			},
+			func(result interface{}) (bool, string, error) {
+				if result == nil {
+					return false, "globalip not found yet", nil
```
Comment (tpantelis) [inline test/e2e/framework/globalingressips.go]:
```suggestion
					return false, fmt.Sprintf("GlobalEgressIP %s not found yet", name), nil
```

**Review by tpantelis — CHANGES_REQUESTED** [review]

File: test/e2e/framework/globalingressips.go [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+	"k8s.io/client-go/dynamic"
+)
+
+var globalIngressIPGVR = &schema.GroupVersionResource{
+	Group:    "submariner.io",
+	Version:  "v1",
+	Resource: "globalingressips",
+}
+
+func (f *Framework) AwaitGlobalnetIP(cluster ClusterIndex, name, namespace string) string {
```
Comment (vthapar) [inline test/e2e/framework/globalingressips.go]: You should have an `AwaitGlobalIngressIP` which returns the object itself. You may need it for tests that test different conditions. I used just IP coz that is more relevant for LH. Add a separate AwaitGlobalnetIP that uses this new method to check for if IP is present or not. That way, you can also test use cases which require a globelIngressIP but allocatedIP to be empty.

**Review by vthapar — CHANGES_REQUESTED** [review]

File: test/e2e/framework/globalingressips.go (hunk 4 above)
Comment (sridhargaddam) [reply test/e2e/framework/globalingressips.go]: Can you tell me how I can test a e2e scenario where GlobalIngressIP Object is created but IPAddress is not allocated based on the current implementation?

Comment (vthapar) [reply test/e2e/framework/globalingressips.go]: IP allocation failed. You can cause such a failure by creating an EgressIPObject that uses up all the remaining IPs in the pool.

Comment (sridhargaddam) [reply test/e2e/framework/globalingressips.go]: As discussed we are validating this in our unit tests and e2e may not be appropriate

**Comment by sridhargaddam** [comment]
Addressed the review comments @vthapar @tpantelis

**Review by tpantelis — APPROVED** [review]

**Review by vthapar — APPROVED** [review]
